Library/Homebrew/cleanup.rb

```diff
@@ -595,6 +595,10 @@ def self.autoremove(dry_run: false)
 
       removable_formulae = Formula.unused_formulae_with_no_dependents(formulae, casks)
 
+      if include_build? || Homebrew::EnvConfig.autoremove_include_build?
+        removable_formulae -= Formula.installed.flat_map(&:build_dependencies)
+      end
```

Comment:
This seems to do the opposite of what you intend it to. Though, I'm also confused about what your intentions are here, so maybe my assessment here is mistaken.

Reply:
No, I'm sure I'm wrong. I'm still trying to understand the architecture, and ruby syntax is still puzzling to me.